rgbd/CMakeLists.txt

```diff
@@ -98,12 +99,30 @@ target_link_libraries(viewer rgbd)
 
 # - - - - - - - - - - - - - - - - - TESTS - - - - - - - - - - - - - - - - -
 
-add_executable(rgbd_test_server test/test_server.cpp)
+add_executable(rgbd_test_server test/test_server.cpp test/test_server_templ.h)
```

Comment:
Nice re-use of code. Did you try using gtest + rostest for automated testing? Would be a nice example.